consensus/types/src/data_column_sidecar.rs

```diff
@@ -23,13 +24,47 @@ pub type ColumnIndex = u64;
 pub type Cell<E> = FixedVector<u8, <E as EthSpec>::BytesPerCell>;
 pub type DataColumn<E> = VariableList<Cell<E>, <E as EthSpec>::MaxBlobCommitmentsPerBlock>;
 
-/// Container of the data that identifies an individual data column.
-#[derive(
-    Serialize, Deserialize, Encode, Decode, TreeHash, Copy, Clone, Debug, PartialEq, Eq, Hash,
-)]
-pub struct DataColumnIdentifier {
+/// Identifies a set of data columns associated with a specific beacon block.
+#[derive(Encode, Clone, Debug, PartialEq)]
+pub struct DataColumnsByRootIdentifier {
     pub block_root: Hash256,
-    pub index: ColumnIndex,
+    pub columns: RuntimeVariableList<ColumnIndex>,
+}
+
+impl RuntimeVariableList<DataColumnsByRootIdentifier> {
+    pub fn from_ssz_bytes_with_nested(
```

Comment:
I think we can simplify this by removing the logic to decode the variable list and just reuse `RuntimeVariableList::from_ssz_bytes`